docs/lib/manual/Flash.md

```diff
@@ -0,0 +1,142 @@
+# Using non-volatile storage from uVisor on mbed OS
+
+This guide will help you to read and write from non-volatile storage in a secure box protected by uVisor on mbed OS 5. If you haven't done so, first [set up an example uVisor program](../QUICKSTART.md). This guide is applicable for the NXP FRDM-K64F development board, but the same principles apply to other development boards.
+
+## Adding the FreescaleIAP library
+
+A quick way to find any issues with running your code under uVisor is by writing code as of uVisor is not present. If any issues arise, uVisor will print them out during [startup](../DEBUGGING.md). The de-facto library to deal with the non-volatile storage on Kinetes boards is the [FreescaleIAP](https://developer.mbed.org/users/Sissors/code/FreescaleIAP/) library by [Erik Olieman](https://developer.mbed.org/users/Sissors/). Add this library by running the following command from your project folder:
```

Comment:
Query: Should "as of uVisor" be "as if uVisor"?

Please delete the hyphen in "nonvolatile".